src/pathfinder/src/main.cc

```diff
@@ -26,41 +28,132 @@ struct CtlMsg
 class Pathfinder
 {
   std::vector<float> _angles;
+  std::vector<float> _filtered;
+#define lidar_steering
 
 public:
   Pathfinder() {}
 
   CtlMsg update_frame(sensor_msgs::LaserScan const& frame)
   {
-    if ((_angles.size() == 0) || (_angles[0] != frame.angle_min))
+    float _ranges = frame.ranges;
+    float _angle_min = frame.angle_min;
+    float _angle_max = frame.angle_max;
+    float _angle_increment = frame.angle_increment;
+    float _range_max = frame.range_max;
+    float _range_min = frame.range_min;
+
+    if ((_angles.size() == 0) || (_angles[0] != _angle_min[0]))
+    {
+      _angles.push_back(_angle_min[0]);
+      for (size_t i = 1; i < _ranges.size(); i++)
+        _angles.push_back(_angles.back() + _angle_increment[0]);
+    }
+
+    assert(_angles.size() == _ranges.size());
+
+#ifdef lidar_steering
+    for (size_t i = 0; i < _angles.size(); i++)
+    {
+      _filtered.push_back(_ranges[i]);
+    }
+
+    // Safety radius for car
+    float safety = 0.500 / 2.0;
+
+    for (int i = 0; i < (int)(_angles.size()); i++)
```

Comment:
These parens are unnecessary.
```suggestion
    for (int i = 0; i < (int)_angles.size(); i++)
```